ibis/expr/table_method.py

```diff
@@ -0,0 +1,75 @@
+import functools
+import inspect
+
+
+class bound_table_method:
```

Comment:
This looks like a mixin of a more generic proxy type (like [this one](https://github.com/PEAK-Legacy/ProxyTypes#proxy-basics)) and a method. 
We already have deferred which also acts like a proxy type so it could make sense to introduce a generic proxy class and use that both here and for deferred.